vyper/parser/memory_allocator.py

```diff
@@ -0,0 +1,24 @@
+from typing import (
+    Tuple,
+)
+
+from vyper.utils import (
+    MemoryPositions,
+)
+
+
+class MemoryAllocator():
+    next_mem: int
+
+    def __init__(self, start_position: int = MemoryPositions.RESERVED_MEMORY):
+        self.next_mem = start_position
+
+    # Get the next unused memory location
+    def get_next_memory_position(self) -> int:
+        return self.next_mem
+
+    # Grow memory by x bytes
+    def increase_memory(self, size: int) -> Tuple[int, int]:
+        before_value = self.next_mem
+        self.next_mem += size
```

Comment:
Now that this is encapsulated in a function, we can double check that it's 32 aligned.
```suggestion
        self.next_mem += size
        assert self.next_mem % 32 == 0
```

Reply:
Good idea :+1:

Reply:
We've tended to avoid using assertions in runtime code because they are easily disable.

https://github.com/ethereum/snake-charmers-tactical-manual/issues/22

Reply:
Maybe we should add a `CompilerPanic` exception? This could also automate adding 'please file an issue' to the exception message.

Reply:
@pipermerriam yip already did it with an exception ;)